File: Easy/94. Binary Tree Inorder Traversal/recursive-solution.js
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function (root) {
  let inorder = [];
  traverse(root);
  return inorder;

  function traverse(node) {
    if (node == null) return;

    traverse(node.left);
    inorder.push(node.val);
    traverse(node.right);
  }
};

File: Easy/94. Binary Tree Inorder Traversal/RecursiveSolution.java
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> result = new ArrayList<>();
        inorder(root, result);
        return result;
    }

    private void inorder(TreeNode node, List<Integer> list) {
        if (node == null)
            return;

        inorder(node.left, list);
        list.add(node.val);
        inorder(node.right, list);
    }
}